source/docs/software/hardware-apis/misc/addressable-leds.rst

```diff
@@ -107,50 +187,678 @@ LEDs can be set with the ``setHSV`` method that takes 4 arguments: index of the
+      ```
+
+   .. tab-item:: C++
+      :sync: C++
+
+      ```C++
+      // Create an LED pattern that displays a red-to-blue gradient, then scroll at one quarter of the LED strip's length per second.
+      // For a half-meter length of a 120 LED-per-meter strip, this is equivalent to scrolling at 12.5 centimeters per second.
+      std::array<Color, 2> colors{Color::kRed, Color::kBlue};
+      LEDPattern base = LEDPattern::DiscontinuousGradient(colors);
+      LEDPattern pattern = base.ScrollAtRelativeSpeed(units::hertz_t{0.25});
+      LEDPattern absolute = base.ScrollAtAbsoluteSpeed(0.125_mps, units::meter_t{1/120.0});
+
+      // Apply the LED pattern to the data buffer
+      heightDisplay.ApplyTo(m_ledBuffer);
+
+      // Write the data to the LED strip
+      m_led.SetData(m_ledBuffer);
+      ```
+
+#### Breathe
+
+.. video:: images/breathe.webm
+   :loop:
+   :poster: ../../../../_images/discontinuous-gradient.png
+   :width: 900
+
+A breathing modifier will make the base pattern brighten and dim in a sinusoidal pattern over the given period of time. Brightness is relative to the original brightness of the base pattern - breathing will only make it dimmer, never brighter than the original.
+
+.. tab-set::
+
+   .. tab-item:: Java
+      :sync: Java
+
+      ```Java
+      // Create an LED pattern that displays a red-to-blue gradient, breathing at a 2 second period (0.5 Hz)
+      LEDPattern base = LEDPattern.discontinuousGradient(Color.kRed, Color.kBlue);
+      LEDPattern pattern = base.breathe(Seconds.of(2));
+
+      // Apply the LED pattern to the data buffer
+      pattern.applyTo(m_ledBuffer);
+
+      // Write the data to the LED strip
+      m_led.setData(m_ledBuffer);
+      ```
+
+   .. tab-item:: C++
+      :sync: C++
+
+      ```C++
+      // Create an LED pattern that displays a red-to-blue gradient, breathing at a 2 second period (0.5 Hz)
+      std::array<Color, 2> colors{Color::kRed, Color::kBlue};
+      LEDPattern base = LEDPattern::DiscontinuousGradient(colors);
+      LEDPattern pattern = base.Breathe(2_s);
+
+      // Apply the LED pattern to the data buffer
+      heightDisplay.ApplyTo(m_ledBuffer);
+
+      // Write the data to the LED strip
+      m_led.SetData(m_ledBuffer);
+      ```
+
+#### Blink
+
+.. video:: images/blink-symmetric.webm
+   :loop:
+   :poster: ../../../../_images/discontinuous-gradient.png
+   :width: 900
+
+.. video:: images/blink-asymmetric.webm
+   :loop:
+   :poster: ../../../../_images/discontinuous-gradient.png
+   :width: 900
+
+Blinking can be done in one of three ways:
+
+1. Symmetrically, where an equal amount of time is spent in the "on" and "off" states per cycle
+2. Asymetrically, where the time spent "on" can be configured independently from the time spent "off"
+3. Synchronously, where the time spent on and off is synchronized with an external source (for example, the state of the RSL)
+
+.. note:: The time-based blinking modifiers are synced with the robot's clock
```

Comment:
At first I wasn't sure what this meant. After reading the note in modifying effects, it makes more sense, but since that note comes later, maybe more of it should be up here.